journal.py

```diff
@@ -51,36 +67,83 @@ def all(cls, session=None):
             session = DBSession
         return session.query(cls).order_by(cls.date.desc()).all()
 
+    @classmethod
+    def search(cls, id, session=None):
+        if session is None:
+            session = DBSession
+        return session.query(cls).filter_by(id=id).one()
+
+    @classmethod
+    def update(cls, id, title=None, text=None, session=None):
+        if session is None:
+            session = DBSession
+        entry = session.query(cls).filter_by(id=id).one()
```

Comment:
remember that this will raise an exception if there is no entry in your database with the give `id`.  How will your application handle that error if it is raised?